### ISSUE TITLE
CMake INSTALL target failes for the RelWithDebInfo configuration on Windows because it cannot find "Stepmania.vdi"

### ISSUE THREAD

**Issue by Hay1tsme** [stepmania/stepmania#2101]
Compiling on Windows 10 with Visual Studio 2019, on a fresh 5_1-new branch clone.

```
19>CMake Error at src/cmake_install.cmake:72 (file):
19>  file INSTALL cannot find
19>  "C:/Users/(username)/Desktop/Games/stepmania/Program/StepMania.vdi": No error.
19>Call Stack (most recent call first):
19>  cmake_install.cmake:38 (include)
```

A file exists at that directory named "StepMania-release-symbols.vdi" but nothing that's just named "StepMania.vdi"
Renaming the file allowed the INSTALL target to finish properly.

**Comment by geefr** [comment]
Hookay:
(Do you mean VS2019? even year releases stopped existing after 2012 I think)

You shouldn't edit cmake_install.cmake - that's a generated file. Correct place for the fix would be src/CMakeLists.txt:754.

The command to generate it is set at line 212 - runs mapconv ${SMNAME_XXX}.map ${SMNAME_XXX}.vdi.

And the definition of the names is at line 57:
```
  set(SM_NAME_RELEASE "StepMania")
  set(SM_NAME_DEBUG "StepMania-debug")
  set(SM_NAME_MINSIZEREL "StepMania-min-size")
  set(SM_NAME_RELWITHDEBINFO "StepMania-release-symbols")
```

So basically the fix is to update line 754 to use the correct filename, probably replace the abuse of generator expressions/separate variables at 57 & 212.

The name you got means you're building in relwithdebinfo configuration, so really the bug is 'make install wont work on windows in non-release builds' or similar.

Phew :)

**Comment by Hay1tsme** [comment]
Whoops! I meant 2019 but I typo'd, apologies!

I didn't editing cmake_install.cmake, I renamed the "Stepmania-release-symbols.vdi" file to "Stepmania.vdi" so that the INSTALL project would finish properly.

How would I update 754 with the correct file name? Apologies but I'm a CMake novice

**Comment by geefr** [comment]
Gotcha. tbh if you're a novice with cmake you might struggle with stepmania, big project with lots of pitfalls.

Main bit is changing the install command, and I'd say preferably without copying the generator expressions from the mapconv run
(Mostly a style point, but there's no need for the $<> syntax there over just making a general SM_NAME variable for the current build type).

What's worse here is mapconv generates the .vdi inside the stepmania source - Means that if you've done a release build first then the relwithdebinfo or other config will actually install the .vdi from the release config instead of their own.
(In general any generated files should be inside cmake's build directory, not thrown into the source code)

Ideal case would be someone volunteers a few weeks on the cmake project, I'll give that one a pass for now and not sure who else would be up to the challenge :/



TLDR/Point fix: Change line 754 to the huge "$<>$<>" string at line 212. When the project is generated that'll be converted to the correct file name for whatever the current build type is (As long as you only build the 4 standard configs)

**Comment by Hay1tsme** [comment]
Understood, just confirmed that it works, thanks for the help! Maybe someday if nobody has cleaned it up by the time I get more comfortable with cmake I'll take up the challenge, haha

**Comment by coderjo** [comment]
It's making it in the program directory, where the exe and map files are output by the compiler, iirc

**Comment by geefr** [comment]
Yes it is, should really be CMAKE_BUILD_DIR/something so it's not shared across multiple builds (Not a big issue though, maybe not worth the effort to fix)

**Comment by Hay1tsme** [comment]
What are the vdi files anyway? Would it be easy to change where it builds them? Seems like a 1 or 2 line fix...

**Comment by coderjo** [comment]
The format comes from VirtualDub, but they contain the addresses of the various functions in the program (from the map file the compiler outputs) so that the crash handler can build a stack backtrace. IIRC, The crash handler expects to find it in the same directory as the program executable (which is built into the Program directory so that running the build output can find the assets such as the theme and noteskins)

**Comment by Hay1tsme** [comment]
Does the crash handler look for a specific file name, or just any .vdi file?

**Comment by coderjo** [comment]
It looks for one named the same as the exe file.

https://github.com/stepmania/stepmania/blob/5_1-new/src/archutils/Win32/CrashHandlerChild.cpp#L68
